ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -171,10 +216,30 @@ protected:
         return Eigen::MatrixX<ScalarType>{};
     }();
 
+    auto strainFunction      = underlying().strainFunction(par, dx);
+    const auto C             = underlying().materialTangentFunction(par);
+    const auto geo           = underlying().localView().element().geometry();
+    const auto numberOfNodes = underlying().numberOfNodes();
+
     auto calculateMatrixContribution = [&]<typename EAST>(const EAST& easFunction) {
       typename EAST::DType D;
       calculateDAndLMatrix(easFunction, par, D, LMat, dx);
 
+      for (const auto& [gpIndex, gp] : strainFunction.viewOverIntegrationPoints()) {
+        const auto M            = easFunction.calcM(gp.position());
+        const double intElement = geo.integrationElement(gp.position()) * gp.weight();
+        const auto EVoigt       = strainFunction.evaluate(gpIndex, on(gridElement)).eval();
+        const auto CEval        = C(gpIndex);
```

Comment:
I am not sure what is correct here:
We are evaluating $\mathbb{C}$ here with the displacements only. And then $\mathbf{S} = \mathbf{CM}\alpha$ with the $\alpha$s only. 
What I have done is obtaining the stress and material tangent by
$\mathbf{E} = \mathbf{E}^c + \mathbf{E}^{EAS}$.

Reply:
Don't we do the same where the contribution related to $\mathbf{E}^C$ is evaluated in `nonlinearelastic.hh` or `linearelastic.hh`? This can be done due to the additive split of $\mathbf{E}$?

Reply:
I just got what you meant, I think it has to be checked in the literature. We can discuss this. I am not sure at the moment if the strain energy function used to define the material law uses $\mathbf{E}$ or $\mathbf{E}^C$? The implementation assumes $\mathbf{E}^C$

Reply:
Yes I think you are right!
The resulting stiffness matrix is the same for $E^c$ or $E^c + $E^{EAS}$

Reply:
This point is revisited [here](https://github.com/ikarus-project/ikarus/pull/325#discussion_r1822300542) and hence marking this conversation as resolved.